### DISCUSSION TITLE
Impose boundaries for a triangulation via geometries instead of enforced edges

### DISCUSSION THREAD

**Discussion by andreapiacentini** [artem-ogre/CDT#150] [Ideas]
Good morning,
we are evaluating CDT for generating a finite element mesh connecting observation locations for ocean surface temperature.
Accordingly to the kind of satellite we can have clouds of points or along track observations locations.
The relevant information is limited to the ocean and should not connect different basins or even different satellite tracks.
We are indeed needing a constrained triangulation, with boundaries and hole, but we do not know which vertices are to be considered as imposed edges.
It would be more natural to describe a geometry that provides the land boundaries or the tracks envelopes and ask the triangulation to cull the triangles whose edges would cross the imposed boundary lines. The actual mesh outer edge would be internal to our limiting geometries. I attach two images for clarifying the situation. I hope I have been clear enough.
Thank you
Andrea

<img width="1009" alt="Screenshot 2023-10-20 at 08 41 45" src="https://github.com/artem-ogre/CDT/assets/316412/d143837c-ec8c-4dca-9103-82cf5edaadf8">

<img width="853" alt="Screenshot 2023-10-20 at 08 34 26" src="https://github.com/artem-ogre/CDT/assets/316412/4c56e81e-6ada-4dcc-a9d3-5ecd8defbc75">

**Comment by artem-ogre** [comment]
Hi, @andreapiacentini 

If I understood you right:
- you have geometry that correctly describes land/water boundaries
- you have a point cloud of observations that you want to include into the triangulation

In that case what you can do is to combine boundary points and observation points in a single buffer (make sure there are no exact duplicates). You can then insert boundary edges as constraints and remove outer triangles and holes.

**Reply by andreapiacentini** [reply]
Thanks.
Sorry for my ignorance, but with your proposed solutions, aren't the geometry boundaries going to be in the final mesh ?
In what I need, the mesh has to stop at the red triangles, not including the green layer that would come from the addition of the coast segments (here South Cornwall to situate where the land is). The only allowed vertices are the ones from the observation location.
![SST_OBS_in_i5ou_20190505_20190509_sstfb_01_fdbk](https://github.com/artem-ogre/CDT/assets/316412/e306617d-8707-4019-8fa2-0e5c065d5c13)

**Reply by artem-ogre** [reply]
Maybe you could add a post-processing step that manually removes all boundary points and all triangles adjacent to them.
